numpy/core/src/common/simd/neon/reorder.h

```diff
@@ -53,6 +53,54 @@
     #define npyv_combineh_f32(A, B) vcombine_f32(vget_high_f32(A), vget_high_f32(B))
 #endif
 
+#define _MM_SHUFFLE(fp3,fp2,fp1,fp0) (((fp3) << 6) | ((fp2) << 4) | \
+                                     ((fp1) << 2) | ((fp0)))
+
+NPY_FINLINE float32x4_t npy_shuffle_ps_1032(float32x4_t a, float32x4_t b)
+{
+    float32x2_t a32 = vget_high_f32(vreinterpretq_f32_m128(a));
+    float32x2_t b10 = vget_low_f32(vreinterpretq_f32_m128(b));
+    return vreinterpretq_m128_f32(vcombine_f32(a32, b10));
+}
+
+NPY_FINLINE float32x4_t npy_shuffle_ps_2301(float32x4_t a, float32x4_t b)
+{
+    float32x2_t a01 = vrev64_f32(vget_low_f32(vreinterpretq_f32_m128(a)));
+    float32x2_t b23 = vrev64_f32(vget_high_f32(vreinterpretq_f32_m128(b)));
+    return vreinterpretq_m128_f32(vcombine_f32(a01, b23));
+}
+
+NPY_FINLINE float32x4_t npy_shuffle_ps_default(float32x4_t a, float32x4_t b, int imm8)
+{
+    float32x4_t ret;
+    ret = vmovq_n_f32(vgetq_lane_f32(vreinterpretq_f32_m128(a), (imm) & 0x3));
+    ret = vsetq_lane_f32(vgetq_lane_f32(vreinterpretq_f32_m128(a), ((imm) >> 2) & 0x3), ret, 1);
+    ret = vsetq_lane_f32(vgetq_lane_f32(vreinterpretq_f32_m128(b), ((imm) >> 4) & 0x3), ret, 2);
+    ret = vsetq_lane_f32(vgetq_lane_f32(vreinterpretq_f32_m128(b), ((imm) >> 6) & 0x3), ret, 3);
+    vreinterpretq_m128_f32(ret);
```

Comment:
```suggestion
    return vreinterpretq_m128_f32(ret);
```
This code clearly isn't exercised - is there some way we can test it?

Reply:
Will integrate into #16641 for test purpose If consensus is reached. This PR is just an idea incubator.